src/api/form/PDFForm.ts

```diff
@@ -499,6 +508,68 @@ export default class PDFForm {
     return PDFTextField.of(text, text.ref, this.doc);
   }
 
+  /**
+   * Flatten all form fields.
+   * 
+   * Flattening a form field will take the current appearance and make that part
+   * of the pages content stream. All form fields and annotations associated are removed.
+   * 
+   * For example:
+   * ```js
+   * const form = pdfDoc.getForm();
+   * form.flatten();
+   * ```
+   */
+  flatten() {
+    this.updateFieldAppearances();
```

Comment:
In most cases, users will want the field appearances to be automatically updated. But this won't _always_ be the case, so we should make sure users can opt out of this behavior. The best way to handle this is probably to add a `updateFieldAppearances` option to this method, similar to `pdfDoc.save(...)`: https://github.com/Hopding/pdf-lib/blob/b169e2fda05beacfe722ed1b3e73e4ec9a59a589/src/api/PDFDocument.ts#L1223